EthersJs/accelerating-development-with-ethersjs.md

```diff
@@ -0,0 +1,242 @@
+
+To get started, download the following initial [Angular application](https://github.com/jacobcreech/Ethersjs-example/tree/initial). To start, run
+
+```
+npm install
+ng serve --open
+```
+
+You should be greeted with the words "initial application" in your browser.
+
+Ethersjs follows the general standard of installing packages for your application. Simply run
+
+```
+npm install --save ethers
+```
+
+Now everything is set up to work with Ethersjs on our simple application.
+
+
+## Creating a Wallet
+
+We are first going to use Ethersjs to create a new wallet. Change to the following html:
+
+/src/app/wallet/wallet.component.html
+
+```
+<div fxFlex="20"></div>
+<div fxFlex="60" class="wallet">
+  <button mat-raised-button color="primary" (click)="onSubmit()">Generate Wallet</button>
+  <p *ngIf="publickey">Public Key: {{publickey}}</p>
+  <p *ngIf="privatekey">Private Key: {{privatekey}}</p>
+</div>
+<div></div>
+```
+
+To create a wallet, we use the wallet.createRandom() to create a random public and private key. We can use this wallet to do other actions with, such as creating transactions and signing messages.
+
+/src/app/wallet/wallet.component.ts
+
+```
+onSubmit() {
+      const randomWallet = ethers.Wallet.createRandom();
+      this.publickey = randomWallet.address;
+      this.privatekey = randomWallet.privateKey;
+      const wallet = new ethers.Wallet(this.privatekey);
+  }
+```
+
+Before we can do anything with this wallet, we must first connect it to the Ethereum blockchain. We will be doing this using the default web3 provider by Ethers.
+
+Add the following to the wallet.component.ts:
+
+```
+provider: any;
+
+ngOnInit() {
+  this.provider = ethers.getDefaultProvider('homestead');;
+}
+```
+
+This obtains the web3 connection provided by Metamask and makes it ready to use by Ethers.
+
+### Sending and Signing Transactions
+
+Next, we create the ability to send a transaction on our application. Ethers provides the ability to edit any data within a transaction, such as gas limit and what address you are sending the transaction to. In order to send a transaction using the wallet, create a send transaction button that uses the sendTransaction method from Ethersjs, as well as some form fields for input on the transaction.
+
+/src/app/wallet/wallet.component.html
+
+```
+<form [formGroup]="transactionForm" (ngSubmit)="sendTransaction(transactionForm.value)">
+        <mat-form-field appearance="outline">
+        <mat-label>To</mat-label>
+        <input formControlName="toAddress" matInput placeholder="0xAddress">
+      </mat-form-field>
+      <mat-form-field appearance="outline">
+        <mat-label>Amount</mat-label>
+        <input formControlName="etherAmount" matInput placeholder="1">
+        <span matSuffix>Ether</span>
+      </mat-form-field>
+        <button mat-raised-button color="secondary">Send Transaction</button>
+    </form>
+```
+
+In your class, import the formModules and create the base form information.
```

Comment:
```suggestion
In the `wallet.component.ts` class, import the `formModules` and create the base form information.
```

People may not fully know Angular.

Also, this step suddenly adds a lot more to the wallet.component.ts file, should we mention all of what you've added here, or change the file in the initial repo, or something else?

Reply:
I'm trying to cover all of the changes to module.ts in the initial repo so that the only work that a reader has to do is in wallet.component.ts. This skips any of the information or changes to Angular that I can avoid, and keeps the general focus to what Ethers.js can do.